backend/generalhw.pm

```diff
@@ -45,20 +45,15 @@ sub run_cmd ($self, $cmd, @extra_args) {
 
     push @full_cmd, @extra_args;
 
-    my ($stdin, $stdout, $stderr, $ret);
-
-    {
-        # Do not let the SIGCHLD handler of Mojo::IOLoop::ReadWriteProcess::Session steal the exit code from IPC::Run
-        local $SIG{CHLD};
-        eval { $ret = IPC::Run::run(\@full_cmd, \$stdin, \$stdout, \$stderr) };
-        die "Unable to run command '@full_cmd' (deduced from test variable $cmd): $@\n" if $@;
-    }
-    chomp $stdout;
-    chomp $stderr;
+    bmwqemu::diag("Calling $cmd");
+    my $ret = _system(@full_cmd);
+    die "Failed to run command '@full_cmd' (deduced from test variable $cmd): $ret\n" if ($ret != 0);
+}
 
-    die "$cmd: stdout: $stdout, stderr: $stderr" unless $ret;
-    bmwqemu::diag("$cmd: stdout: $stdout, stderr: $stderr");
-    return $stdout;
+sub _system (@cmd) {
```

Comment:
You'll need to mark the function as non-coverable. Maybe this works best if you make it a one-liner.